src/as-release.c

```diff
@@ -706,6 +788,8 @@ as_release_legacy_get_default_artifact (AsRelease *release)
 	AsReleasePrivate *priv = GET_PRIVATE (release);
 	g_autoptr(AsArtifact) artifact = NULL;
 
+	g_assert (AS_IS_RELEASE (release));
```

Comment:
I don't think this check is needed here: It is used in a static and internal function, so the input will already have been verified, and a hard `g_assert` will be compiled out in all builds that aren't debug builds anyway.

Reply:
I feel that it is fine to have the g_assert in static functions to know early when something goes wrong without impacting the final release build, but I'll remove it from here